datasources/src/commonMain/kotlin/com/mirego/trikot/datasources/BaseDataSourceV2.kt

```diff
@@ -0,0 +1,143 @@
+package com.mirego.trikot.datasources
+
+import com.mirego.trikot.datasources.extensions.data
+import com.mirego.trikot.foundation.concurrent.AtomicReference
+import com.mirego.trikot.streams.cancellable.CancellableManager
+import com.mirego.trikot.streams.reactive.BehaviorSubject
+import com.mirego.trikot.streams.reactive.Publishers
+import com.mirego.trikot.streams.reactive.distinctUntilChanged
+import com.mirego.trikot.streams.reactive.filter
+import com.mirego.trikot.streams.reactive.filterNotNull
+import com.mirego.trikot.streams.reactive.first
+import com.mirego.trikot.streams.reactive.map
+import com.mirego.trikot.streams.reactive.processors.combine
+import com.mirego.trikot.streams.reactive.promise.Promise
+import com.mirego.trikot.streams.reactive.shared
+import com.mirego.trikot.streams.reactive.subscribe
+import org.reactivestreams.Publisher
+
+private typealias NullableDataState<T> = BaseDataSourceV2.NullableValue<DataState<T, Throwable>>
+private typealias ReadPublisher<T> = Publisher<NullableDataState<T>>
+private typealias DataPublisher<T> = BehaviorSubject<NullableDataState<T>>
+
+abstract class BaseDataSourceV2<R : DataSourceRequest, T>(private val cacheDataSource: DataSource<R, T>? = null) : DataSource<R, T> {
```

Comment:
Can we find a name where it explain why it differenciate with BaseDataSource

Is it only the promises ?

Reply:
PromiseAwareDataSource ?